### ISSUE TITLE
Support ItemKeyedDataSource style paging

### ISSUE THREAD

**Issue by Glurt** [xmartlabs/fountain#53]
This might be an issue with my understanding of the library but I'm struggling to see how I can use it to page an endpoint that doesn't use page numbers. Take Reddit for example, if I want a Listing I have to provide the id of the last element in the previous Listing.

Currently I only have access to `fun fetchPage(page: Int, pageSize: Int): Deferred<T>` which doesn't provide enough information for me to request the next page. Could you add support for a PageFetcher that provides the item given in the `onItemAtEndLoaded` callback?

**Comment by FerrariBruno** [comment]
Yes, the library is definitely focused on Integers at this point. To achieve what you need in the best possible way, we thought you could implement a base class that would use a generic type and then implement the `Integer` and `String` logic. You'll also have a problem with the autoincrement feature for the pages, so that's another thing to look into.
